### PR TITLE
PVP-188

### PR DESCRIPTION
**Pull request by KhajiitLikeToSneak**
- The steps and calories counter has been transferred from the day box to the activities box, with the day of the month now being shown instead.
- User can see counters and completed tasks number related to a specific task type in the box below the current day.
- Activities box is visible only for the current day and any preceding days when navigating the weekly view.
- When user clicks on the day box, the activities box become the tasks box along with filters, and vice versa.

Capture

### CODE REVIEW

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 1]
```diff
@@ -362,10 +351,198 @@ fun Day(
...
+        )
+    ) {
+        Column(
+            modifier = Modifier.fillMaxSize(),
+        ) {
+            ActivityBox(
+                modifier = Modifier.fillMaxWidth(),
+                columnModifier = Modifier.padding(16.dp)
+            ) {
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:385]: Redundant line

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 2]
```diff
@@ -362,10 +351,198 @@ fun Day(
...
+                Text(
+                    fontSize = 18.sp,
+                    modifier = Modifier.align(Alignment.CenterHorizontally),
+                    text = "Todays tasks"
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:389]: Today's

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 3]
```diff
@@ -362,10 +351,198 @@ fun Day(
...
+                )
+
+                Divider(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:392]: Deprecated, no? HorizontalDivider/VerticalDivider are alternatives

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 4]
```diff
@@ -362,10 +351,198 @@ fun Day(
...
+                        text = "Steps of the day",
+                        textAlign = TextAlign.Center,
+                        style = MaterialTheme.typography.titleSmall,
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(top = 4.dp)
+                    )
+
+                    StepCounter(date = date)
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:437]: Redundant line

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 5]
```diff
@@ -362,10 +351,198 @@ fun Day(
...
+                        CalorieCounter(date = date)
+                    }
+
+                    ActivityRow {
+                        Icon(
+                            imageVector = Icons.Outlined.MonitorHeart,
+                            contentDescription = null,
+                            modifier = Modifier.size(26.dp)
+                        )
+                        Text(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:457]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 6]
```diff
@@ -362,10 +351,198 @@ fun Day(
...
+                        )
+                    }
+
+                    ActivityRow {
+                        Icon(
+                            imageVector = Icons.Outlined.Nightlight,
+                            contentDescription = null,
+                            modifier = Modifier.size(26.dp)
+                        )
+                        Text(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:470]: Same as above

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 7]
```diff
@@ -362,10 +351,198 @@ fun Day(
...
+                            text = "0 hr 0 m",
+                            style = MaterialTheme.typography.titleSmall,
+                            modifier = Modifier.padding(start = 8.dp)
+                        )
+                    }
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:476]: Redundant line

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 8]
```diff
@@ -362,10 +351,198 @@ fun Day(
...
+@Composable
+fun TodaysTasks(
+    icon: ImageVector,
+    tasks: List<Task>,
+    taskCategory: KClass<out Task>
+) {
+    val tasksOfCategory = tasks.filter { it::class == taskCategory }
+    val completedTasks = tasksOfCategory.filter { it.isCompleted }
+    val text =
+        if (tasksOfCategory.isEmpty()) "-" else "${completedTasks.size}/${tasksOfCategory.size}"
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:492]: Split to make `if` begin within varable's line

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 9]
```diff
@@ -362,10 +351,198 @@ fun Day(
...
+
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Icon(
+            imageVector = icon,
+            contentDescription = null
+        )
+        Text(text = text)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:501]: Split

File: app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt [hunk 10]
```diff
@@ -416,19 +593,21 @@ fun StepCounter(
...
-            val radius = 300f
+            val radius = min(size.width, size.height) / 2 - strokeWidth
             val topLeft = Offset(
-                (size.width / 2) - (radius / 2),
-                (size.height / 2) - (radius / 2)
+                (size.width / 2) - radius,
+                (size.height / 2) - radius
             )
-            val size = Size(radius, radius)
+            val size = Size(radius * 2, radius * 2)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarWeeklyScreen.kt:610]: Params could be split and multi-line variables separated by an empty line.

**Review by klaidoshka — APPROVED** [review]

**Comment by klaidoshka** [comment]
Maybe second row (steps, sleep columns) could have bigger top padding, probably would look better

**Review by vytjar — APPROVED** [review]

**Pull request merged**